openwisp_monitoring/device/admin.py

```diff
@@ -63,10 +64,38 @@ class CheckInline(GenericStackedInline):
     fields = ['check_type', 'is_active']
     readonly_fields = ['check_type']
 
+    @property
+    def check_app(self):
+        if 'SAMPLE_APP' in environ:
+            return 'sample_check'
+        return 'check'
+
+    def _set_check_inline(self):
+        self.fields = []
+        self.readonly_fields = ['created', 'modified']
+
     def has_add_permission(self, request, obj=None):
+        if request.user.has_perm(f'{self.check_app}.add_check_inline'):
```

Comment:
You should be able to get the `model_name` like this: 

```suggestion
        if request.user.has_perm(f'{self.model._meta.model_name}.add_check_inline'):
```

Reply:
I think it should be `if request.user.has_perm(f'{self.model._meta.app_label}.add_check_inline'):`